packages/build-utils/src/get-project-paths.ts

```diff
@@ -0,0 +1,67 @@
+import { detectFramework } from './detect-framework';
+import { DetectorFilesystem } from './detectors/filesystem';
+import frameworks from '@vercel/frameworks';
+
+const MAX_DEPTH_TRAVERSE = 3;
+
+export interface GetProjectPathsOptions {
+  fs: DetectorFilesystem;
+  path?: string;
+  skipPaths?: string[];
+}
+
+export type ProjectPath = string;
+
+export const getProjectPaths = async ({
+  fs,
+  path,
+  skipPaths,
+}: GetProjectPathsOptions): Promise<ProjectPath[] | any> => {
+  const allPaths: Array<ProjectPath> = [];
+  for (
+    let currentDepth = 0;
+    currentDepth < MAX_DEPTH_TRAVERSE;
+    currentDepth++
+  ) {
+    if (path && skipPaths?.includes(path)) {
+      return allPaths;
+    } else {
+      const directoryContents = await fs.readdir(path ?? './');
+
+      // This should only ever return one file bc there should never be more than one package.json in a directory
+      const childFiles = directoryContents.filter(
+        dir => dir.type === 'file' && dir.name === 'package.json'
+      );
+      const childDirs = directoryContents.filter(stat => stat.type === 'dir');
+
+      // Scenario 3
+      if (!childDirs.length && !childFiles.length) {
+        return allPaths;
+      }
+
+      if (childDirs.length) {
+        const dirPaths = await Promise.all(
+          childDirs.flatMap(async current => {
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
`paths` seems to be used. Does this `eslint-disable` need to be here for this file to lint successfully?

Reply:
Let me try removing it but it is actually for the recursive call is what the eslint was for

Reply:
Removed